enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -212,6 +212,59 @@ These changes have already been implemented, and the initial PR for them can be
 
 In order to use `oc` with an external OIDC provider, the tool has been [extended](https://github.com/openshift/oc/pull/1640) with the necessary functionality, including command-line arguments that enable the required configuration. In particular, [`oauth2cli`](https://github.com/int128/oauth2cli) has been vendored into the `oc` codebase. One important consideration here is that depending on the OIDC provider, further functionality might be required, in which case `oc` will have to be extended to support that too.
 
+#### authorization.openshift.io/RestrictSubjectBindings admission plugin considerations
+
+OpenShift's default admission plugin `authorization.openshift.io/RestrictSubjectBindings`
+utilizes the OAuth stack, specifically the `User` and `Group` APIs, to ensure `RoleBinding`
+subjects do not violate any restrictions in the namespace they are going to be created in.
+These restrictions are specified by `RoleBindingRestriction` resources that are served
+by the openshift-apiserver.
+
+Removing the OAuth stack entirely during external OIDC setup may result in the `authorization.openshift.io/RestrictSubjectBindings` admission plugin, and thus the KAS appearing degraded.
+
+##### Changes to the kube-apiserver
+
+To account for the impacts to the `authorization.openshift.io/RestrictSubjectBindings` admission plugin, the OpenShift-specific patch to the kube-apiserver that adds this admission plugin will be updated such that:
+
+- Informers for the `Group` API are not started if the `Authentication` resource `.Spec.Type` is set to `OIDC`
+- The post-start hook that checks for oauth-apiserver connectivity will be skipped if the `Authentication` resource `.Spec.Type` is set to `OIDC`
+- `RoleBinding`s will be rejected if there exists a `RoleBindingRestriction` that specifies user and/or group restrictions
+- It is considered a failure if we are unable to determine the authentication type for the cluster, leading to rejection of the `RoleBinding`
+
+##### Changes to openshift-apiserver
+
+To help keep users informed of the expected behavior of the `authorization.openshift.io/RestrictSubjectBindings` admission plugin when using the OIDC cluster authentication mode, it is proposed that a new admission plugin is added to the openshift-apiserver to reject creation of `RoleBindingRestriction` resources containing user/group restrictions.
```

Comment:
Yeah, I am aware that there is some downside here by adding a new admission plugin to maintain but I think it will make the user experience of using the OIDC authentication mode nicer. 

I don't think it is a "must have" feature, but rather a "nice to have" that makes the OIDC enablement a more holistic change to the cluster behavior.

If we do decide to go the route of requiring no RBRs with user/group specifications in order to enable OIDC, I would say this does become a "must have" to prevent creation/update of resources in a way that would have prevented the enablement of the feature.

Reply:
No longer adding an admission plugin. Instead we are planning to remove the API.